rs/bitcoin/ckbtc/minter/src/state/eventlog.rs

```diff
@@ -335,15 +345,9 @@ pub fn replay(mut events: impl Iterator<Item = Event>) -> Result<CkBtcMinterStat
                 }
             }
             Event::RetrieveBtcKytFailed { kyt_provider, .. } => {
-                // Only charge kyt fee for failed RetrieveBtcRequest when the provider is not the new KYT canister
-                if state
-                    .new_kyt_principal
-                    .map(|canister_id| !kyt_provider.eq(&canister_id.into()))
-                    .unwrap_or(true)
-                {
-                    *state.owed_kyt_amount.entry(kyt_provider).or_insert(0) += state.kyt_fee;
-                }
+                *state.owed_kyt_amount.entry(kyt_provider).or_insert(0) += state.kyt_fee;
             }
+            Event::RetrieveBtcOfacFailed { .. } => {}
```

Comment:
If we don't do anything with the state when replaying the event, why do we need this event in the first place? I would have said we need it to reimburse the user, but that's actually another event (`ScheduleDepositReimbursement`) so maybe we don't need `RetrieveBtcOfacFailed` at all and we can remove it?

Reply:
This `RetrieveBtcOfacFailed` was introduced in commit 027abaaa552e6a324081b93005d21d144f476e88 to address one of your earlier comments about this piece of code:
```
 // uuid and block_index are no longer relevant for the new KYT canister
 let uuid = "".to_string();
 let block_index = 0;
 record_event(&Event::RetrieveBtcKytFailed {
        owner,
        address,
        amount,
        kyt_provider,
        uuid,
        block_index,
    });
```

Considering that this `record_event` will be removed in the [followup PR](https://github.com/dfinity/ic/pull/2559) that addresses XC-233, I suggest to keep the above hack in stead of introducing a new event. WDYT?

Reply:
Maybe I'm missing something, but we don't need this hack at all,  no? What should be done in that case when replaying that event? (I'm suspecting, nothing at all)

Reply:
In this PR we are still doing reimbursement. So it looks necessary to me that `RetrieveBtcKytFailed` needs to be kept in order not to break tests.

Or I can merge #2559 into this one if that is what you prefer.

Reply:
> So it looks necessary to me that RetrieveBtcKytFailed needs to be kept in order not to break tests.

I think this is the part I don't understand yet because in the current commit it was replaced by the event `RetrieveBtcOfacFailed` which was a no-op when being replayed, meaning that it doesn't change the state. So if the tests were green then, the tests should still be green without this hack, no?

Reply:
I see what you mean. You are right, it does look like `RetrieveBtcKytFailed` event is only used for kyt_fee accounting. Since there is no longer a kyt_fee in retrieve_btc, it is safe to not record this event. I've pushed a fix to remove `RetrieveBtcOfacFailed`, please have a look again. Thanks!

Reply:
Thanks a lot for the change, I do feel that this is much cleaner that way!